### DISCUSSION TITLE
Github Api Creating File Error 400

### DISCUSSION THREAD

**Discussion by Kale-Ko** [community/community#21449] [Other Feature Feedback, Questions, & Ideas]
<p>I am writing a program that needs to upload a file to a GitHub repo making use of the API. Using <a href="https://docs.github.com/rest/reference/repos#create-or-update-file-contents" rel="noopener nofollow ugc">this</a> as a reference. It looks like I have done everything correctly, but I keep getting a 400 error <code>Problems parsing JSON</code> upon sending a request.</p>
<p>I have double-checked the repo and file input,<br>
I made sure the content is being encoded to Base64 correctly,<br>
and the token is working on other requests so that is correct</p>
<p>The sha is gotten from sending a get request to the same URL and then taking the sha value of the returned json (<a href="https://docs.github.com/en/rest/reference/repos#get-repository-content" rel="noopener nofollow ugc">get repository content</a>)</p>
<pre><code class="lang-javascript">function uploadFile(repo, file, contents, sha, token) {
    var body = { message: "Update", contents: btoa(contents), sha }

    return fetch("https://api.github.com/repos/" + repo + "/contents" + file, { "method": "PUT", body, "headers": { "Content-Type": "application/json", "Authorization": "token " + token } })
        .then(res =&gt; res.json())
        .then(data =&gt; { return data })
}

uploadFile("My/Repo", "A/File.txt", "Test Contents", "sha", "ghp_..") //https://api.github.com/repos/My/Repo/contents/A/File.txt 400 {message: "Problems parsing JSON", documentation_url: "https://docs.github.com/rest/reference/repos#create-or-update-file-contents"}
</code></pre>
<p>I have tried just sending the raw string.<br>
I also tried turning the json into a string and sending that but that returns <code>{message: "Invalid request.↵↵"content" wasn't supplied.", documentation_url: "https://docs.github.com/rest/reference/repos#create-or-update-file-contents"}</code></p>
<p>Does anyone know what is wrong?</p>

**Comment by mattfbacon** [comment]
<p>Read the error message. <code>"content" wasn't supplied."</code> because you named it <code>contents</code> rather than <code>content</code>.</p>

**Comment by Kale-Ko** [comment]
<aside class="quote no-group" data-username="mattfbacon" data-post="2" data-topic="186242" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 mattfbacon:</div>
<blockquote>
<p>Read the error message. <code>"content" wasn't supplied."</code> because you named it <code>contents</code> rather than <code>content</code>.</p>
</blockquote>
</aside>
<p>Thanks for noticing that, but that is not it, Plus I only got that error when I didn’t pass contents in the body.</p>

**Comment by mattfbacon** [comment]
<aside class="quote no-group" data-username="Kale-Ko" data-post="3" data-topic="186242">
<div class="title">
<div class="quote-controls"></div>
 Kale-Ko:</div>
<blockquote>
<p>I only got that error when I didn’t pass contents in the body.</p>
</blockquote>
</aside>
<p>Naturally, because it’s required…</p>

**Comment by Kale-Ko** [comment]
<p>:man_facepalming:t3: The code I supplied does not generate this errror, only when I turn the json into a string and send it. Also as it said changing contents to content does nothing</p>

**Comment by Kale-Ko** [comment]
<p>Fixed it, had to stringify the json with the content field instead of contents</p>

**Comment by mattfbacon** [comment]
<p>(exactly like I said :upside_down_face:)</p>

**Comment by Kale-Ko** [comment]
<p>Not what you communicated</p>

**Comment by mattfbacon** [comment, marked as answer]
<p>I apologize, that was a bit snarky :pray:</p>
<p>In my first reply I said that you named the field <code>contents</code> instead of <code>content</code> and that’s why it said <code>"content" wasn't supplied.</code>. That’s what I mean by “as I said”.</p>
